util/version/version.go

```diff
@@ -162,14 +162,13 @@ func (v buildIdentifier) compare(o buildIdentifier) int {
 			return 1
 		}
 	} else { // both are strings
-		switch {
-		case v.IdentifierStr < o.IdentifierStr:
-			return -1
-		case v.IdentifierStr == o.IdentifierStr:
+		// In order to support random build indentifiers, like commit hashes,
```

Comment:
In order to surface this logic could you kindly update the comment for the `WithBuildTags` with something similar to:
```
// WithBuildTags modifies the version comparison to also consider build tags
// when comparing versions.
// Performs a standard version compare between a and b. If the versions
// are equal, build identifiers will be used to compare further; precedence for two build
// identifiers is determined by comparing each dot-separated identifier from left to right
// until a difference is found as follows:
// - Identifiers consisting of only digits are compared numerically.
// - Numeric identifiers always have lower precedence than non-numeric identifiers.
// - Identifiers with letters or hyphens are compared only for equality, otherwise, 2 is returned given
//.  that it is not possible to identify if lower or greater (non-numeric identifiers could be random build
//.  identifiers).
```

Other comments should be modified accordingly by adding
```
// 2 == v is different than o (it is not possible to identify if lower or greater).
```

Reply:
Done